### DISCUSSION TITLE
Transparent proxy 502 Bad Gateway error: CONNECT proxy-safebrowsing.googleapis.com:443

### DISCUSSION THREAD

**Discussion by IMnnmd** [mitmproxy/mitmproxy#6167] [Q&A]
Hello everyone!

I'm trying to set up a MitM on my linux machine to intercept and decrypt packets from my MacOS device (MacBook Pro, Monterey).
I've set MitM to transparent mode, routed traffic to it, set all the options listed in tutorial (IP forwarding, certificates etc.)
And it works fine with HTTP, but on HTTPS I get the following error in PCAP where I store the collected packets:

![image](https://github.com/mitmproxy/mitmproxy/assets/52769382/3a35664d-e064-4d4c-b133-e036b739aa99)

Although I get errors here, it worked just fine with my Linux device, so I have no clue how to handle this error.
Could someone help me fix it, please?

**Comment by mhils** [comment]
Receiving a CONNECT request in transparent mode can mean two things:

1. You have explicitly configured an HTTP proxy on the client. You should only do one: setup transparent mode OR explicitly configure a proxy.
2. You have _not_ configured an HTTP proxy on the client: In this case your client still tries to connect to another HTTP proxy. mitmproxy does not support transparently proxying connections to another HTTP proxy yet.
